### PR TITLE
LPS-148696 Add missing deleteDepotEntry(DepotEntry) method in the DepotEntryLocalServiceImpl class to also handle the ResourcePermission deletion - LPS-145396 Subtask

### PR DESCRIPTION
**Pull request by jorgediaz-lr**
This is a subtask of [LPS-145396](https://issues.liferay.com/browse/LPS-145396) where I am handling the correct ResourcePermission deletion in the Liferay code.

In parent [LPS-145396](https://issues.liferay.com/browse/LPS-145396) I am implementing some changes in DataGuardTestRule to detect orphan ResourcePermission records, avoiding the bug in the future and I have detected that if you call the current `deleteDepotEntry(DepotEntry)` method that is implemented in DepotEntryLocalServiceBaseImpl, it is not correctly handling the ResourcePermission deletion.

To solve it, I am adding it to DepotEntryLocalServiceImpl and reusing the current code of `deleteDepotEntry(long)` that is working correctly.

Note: In this LPS, I have already sent a previous PR https://github.com/liferay-lima/liferay-portal/pull/2796 to solve ResourcePermission deletion problems of other Lima team components

Let me know if you have any question

### CODE REVIEW

**Comment by jorgediaz-lr** [comment]
ci:test:sf

**Comment by jorgediaz-lr** [comment]
ci:test:relevant

File: modules/apps/depot/depot-service/src/main/java/com/liferay/depot/service/impl/DepotEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -155,21 +155,29 @@ public DepotEntry addDepotEntry(
...
 
-		return super.deleteDepotEntry(depotEntryId);
+		return super.deleteDepotEntry(depotEntry);
+	}
+
+	@Override
+	public DepotEntry deleteDepotEntry(long depotEntryId)
+		throws PortalException {
+
+		DepotEntry depotEntry = depotEntryPersistence.fetchByPrimaryKey(
```
Comment (adolfopa) [inline modules/apps/depot/depot-service/src/main/java/com/liferay/depot/service/impl/DepotEntryLocalServiceImpl.java:177]: `depotEntry` is used only once; can you inline it?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/depot/depot-service/src/main/java/com/liferay/depot/service/impl/DepotEntryLocalServiceImpl.java (hunk 1 above)
Comment (jorgediaz-lr) [reply modules/apps/depot/depot-service/src/main/java/com/liferay/depot/service/impl/DepotEntryLocalServiceImpl.java]: Hi @adolfopa 

I have updated the commit with the requested change, see: https://github.com/liferay-lima/liferay-portal/pull/3417/commits/dfbdb247fd0487c59d2577b97a147c64e4e1abe9#diff-356ff59c01041287ff92ac5c64090d41c29c0ef8666cf52fe6022721e43f828dR177

**Comment by jorgediaz-lr** [comment]
ci:test:sf

**Comment by jorgediaz-lr** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**